indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -0,0 +1,119 @@
+
+from flask import request, redirect
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy.colors import ColorTuple
+from indico.modules.events.sessions.controllers.management import RHManageSessionsBase, RHManageSessionBase
+from indico.modules.events.sessions.forms import CreateSessionForm
+from indico.modules.events.sessions.views import WPManageSessions
+from indico.modules.events.sessions.util import create_session
+from indico.web.flask.util import url_for
+from indico.web.forms.base import FormDefaults
+from indico.web.util import jsonify_data, jsonify_template
+
+
+class RHSessionsList(RHManageSessionsBase):
+    """Display list of all sessions within the event"""
+
+    def _process(self):
+        active_sessions = self.event.sessions.filter_by(is_deleted=False).all()
+        return WPManageSessions.render_template('management/sessions/sessions_list.html', self.event.as_legacy,
+                                                event=self.event, sessions=active_sessions)
+
+
+class RHAddSession(RHManageSessionsBase):
+    """Add a session to the event"""
+
+    def _process(self):
+        form = CreateSessionForm()
+        if form.validate_on_submit():
+            session_config = dict(is_poster=(form.session_type.data == 'poster'))
+            event_session = create_session(self.event, wtform=form, session_config=session_config)
+            db.session.add(event_session)
+            return jsonify_data()
+        return jsonify_template('events/sessions/management/sessions/add_session.html', form=form)
+
+
+class RHEditSession(RHManageSessionBase):
+    """Edit the session"""
+
+    def _process(self):
+        form = CreateSessionForm(obj=self._get_form_defaults())
+        if form.validate_on_submit():
+            session_config = dict(is_poster=(form.session_type.data == 'poster'))
+            create_session(self.event, wtform=form, session_config=session_config, base_session=self.session)
+            return jsonify_data()
+        return jsonify_template('events/sessions/management/sessions/add_session.html', form=form)
+
+    def _get_form_defaults(self):
+        return FormDefaults(self.session, **{
+            'colors': {'background': self.session.background_color, 'text': self.session.text_color},
+            'session_type': ('poster' if self.session.is_poster else 'standard')
+        })
+
+
+class RHRemoveSession(RHManageSessionsBase):
+    """Remove a single session"""
+
+    def _process(self):
+        session = self.event.sessions.filter_by(id=request.view_args['session_id']).first_or_404()
+        session.is_deleted = True
+        return redirect(url_for('.sessions_list', self.event))
+
+
+class RHRemoveSessions(RHManageSessionsBase):
+    """Remove multiple sessions"""
+
+    def _process(self):
+        session_ids = map(int, request.form.getlist('session_id'))
+        for session in self.event.sessions:
+            if session.id in session_ids:
+                session.is_deleted = True
```

Comment:
This should be an operation, since you also need to disassociate all contributions from the sessions (and session blocks / timetable entries associated with it)